latte/evaluation/metrics.py

```diff
@@ -0,0 +1,80 @@
+from scipy import optimize
+import numpy as np
+import torch
+
+from sklearn.feature_selection import mutual_info_regression
+from sklearn.metrics import mutual_info_score
+from scipy.stats import differential_entropy
+
+
+def mig(Z: np.ndarray, Y: np.ndarray, discrete: bool = False) -> float:
+    """
+    Calculates the MIG score of the learned representations `Z` against the true factors of variation `Y`.
+    Mutual information is calculated with `sklearn`'s `mutual_info_regression` function.
+
+    Args:
+        Z: An `N x k` matrix of learned representations for `N` samples.
+        Y: An `N x d` matrix of true factors of variation for `N` samples.
+        discrete: Whether the distributions of the factors/learned representations are discrete.
+
+    Returns:
+        The MIG score.
+    """
+
+    k, d = Z.shape[1], Y.shape[1]
+
+    # Compute the pairwise mutual information matrix
+    M = np.zeros((k, d))
+    for i in range(k):
+        M[i, :] = mutual_info_regression(Y, Z[:, i])
+
+    # Compute the factor of variation entropies
+    H = np.zeros(k)
+    for j in range(k):
+        # TODO (Anej): Which one to use?
```

Comment:
Why is differential entropy needed in MIG score? Do you use it just to have an upper bound for MI, or also for something else?